lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -385,7 +384,17 @@ defmodule Mix.Compilers.Elixir do
     modules_to_recompile =
       modules_to_recompile ++
         for {:ok, {module, true}} <-
-              Task.async_stream(modules_to_mix_check, &{&1, &1.__mix_recompile__?()},
+              Task.async_stream(
+                modules_to_mix_check,
+                fn mod ->
+                  # Since loading the modules themselves can be expensive,
+                  # we want to do it inside the task too. Modules being deleted
+                  # are uncommon, so this optimizes the common case.
+                  {mod,
+                   Code.ensure_loaded?(mod) and
```

Comment:
Won't that effectively synchronize on code_server?

Reply:
Code loading happens mostly on the caller since Erlang/OTP 26, parallelizing only if there are `on_load` callbacks.